File: toh.py
def TOH(n, src, temp, dest):
    if n == 1:
        print(f"Move disk 1 from {src} to {dest}")
    else:
        TOH(n - 1, src, dest, temp) 
        print(f"Move disk {n} from {src} to {dest}")
        TOH(n - 1, temp, src, dest)

n = int(input("Enter the number of disks: "))
print(f"The sequence of moves for {n} disks:")
TOH(n, 'A', 'B', 'C')

File: toh.c
#include<stdio.h>
void TOH(int n, char src, char temp, char dest) 
{
    if (n == 1)
    {
        printf("Move disk 1 from %c to %c\n", src, dest); 
    }
    else
    {
    TOH(n - 1, src, dest, temp);
    printf("Move disk %d from %c to %c\n", n, src, dest);
    TOH(n - 1, temp, src, dest);
}
}

int main() 
{
    int n;
    printf("Enter the number of disks: ");
    scanf("%d", &n);
    printf("The sequence of moves for %d disks:\n", n);
    TOH(n, 'A', 'B', 'C');

    return 0;
}
